src/workers/sync/sync.ts

```diff
@@ -79,15 +79,36 @@ class Sync extends Process {
       this.consumePullQueue(pullFromLocal, this.local, this.remote),
       this.consumePullQueue(pullFromRemote, this.remote, this.local),
     ]);
+
+    const foldersDeletedInLocal = await this.listDeletedFolders(
```

Comment:
Please explain here why a change on the sync is required and how it affects to the sync process, also, remember to update the Desktop documentation according to this

Reply:
When a folder was deleted, the folder and all its contents were deleted, but with trash, we have to move only the one deleted. Not doing so will make it impossible to restore a folder from the trash since the contents will be still deleted.

So the change is only to get the deleted folders before actually deleting anything and filtering the files that don't need to be deleted because they are in a folder that has been deleted